frame/treasury/src/lib.rs

```diff
@@ -561,13 +553,14 @@ decl_module! {
 		/// # <weight>
 		/// - Complexity: `O(T)` where `T` is the number of tippers.
 		///   decoding `Tipper` vec of length `T`.
-		///   `T` is less than `MAX_TIPPERS_COUNT`, it is charged as maximum and then refund.
+		///   `T` is charged as upper bound given by `ContainsCountUpperBound`.
 		///   The actual cost depends on the implementation of `T::Tippers`.
 		/// - DbReads: `Tips`, `Tippers`, `tip finder`
 		/// - DbWrites: `Reasons`, `Tips`, `Tippers`, `tip finder`
 		/// # </weight>
-		#[weight = 211_000_000 + 1_094_000 * MAX_TIPPERS_COUNT + T::DbWeight::get().reads_writes(3, 3)]
-		fn close_tip(origin, hash: T::Hash) -> DispatchResultWithPostInfo {
+		#[weight = 210_000_000 + 1_100_000 * T::Tippers::count_upper_bound() as u64
```

Comment:
I think we'll want to go with `as Weight` directly, to remove potential for code churn in the future.